docs/learn.md

```diff
@@ -400,7 +400,318 @@ over the way these JavaScript packages are downloaded and bundled.
+`modes`— are referred to as _fields_ of the stanza.
+
+All stanzas are well covered in the Dune documentation site, where we can find
+the reference for the [`library`
+stanza](https://dune.readthedocs.io/en/stable/dune-files.html#library).
+
+Dune is designed to minimize changes in configuration when the project folder
+structure changes, so one can move around the `lib` folder to another place
+inside the project, and all build commands will still keep working without any
+changes in Dune configuration. Very handy!
+
+#### Entry points with `melange.emit`
+
+Libraries are useful to encapsulate behavior and logical components of our
+application, but they won’t produce any JavaScript artifacts on their own.
+
+To generate JavaScript code, we need to define an entry point of our
+application. In the root folder, create another `dune` file:
+
+```bash
+(melange.emit
+ (target app)
+ (libraries lib))
+```
+
+And an `app.ml` file:
+
+```
+let () = Js.log Lib.name
+```
+
+The `melange.emit` stanza is the one that tells Dune to generate JavaScript
+files from a set of libraries and modules. In-depth documentation about this
+stanza can be found on the [Dune
+docs](https://dune.readthedocs.io/en/latest/melange.html).
+
+The file structure of the app should look something like this:
+
+```
+project_name/
+├── _opam
+├── lib
+│   ├── dune
+│   └── lib.ml
+├── dune-project
+├── dune
+└── main.ml
+```
+
+#### Building the project
+
+We can build the project now, which will produce the JavaScript code from our
+sources using the Melange compiler:
+
+```bash
+$ dune build @melange
+```
+
+This command tells dune to build all the targets that have an alias `melange`
+attached to them.
+[Aliases](https://dune.readthedocs.io/en/stable/overview.html#term-alias) are
+build targets that don’t produce any file and have configurable dependencies.
+
+By default, all the targets in a `melange.emit` stanza and the libraries it
+depends on are attached to the `melange` alias. We can define explicit aliases
+though, as we will see below.
+
+If everything went well, we should be able to run the resulting JavaScript with
+Node. As we mentioned while going through its features, Dune places all
+artifacts inside the `_build` folder to not pollute any source folders. So we
+will point Node to the script placed in that folder, to see the expected output:
+
+```bash
+$ node _build/default/app/app.js
+Jane
+```
+
+One thing to note is that we have to look for the `app.js` file inside an `app`
+folder, but we don’t have any such folder in our sources. The reason why this
+folder is created is to support multiple `melange.emit` stanzas in the same
+folder. To support this scenario, Dune will use the `target` field defined in
+the `melange.emit` to place the artifacts generated from a `melange.emit` stanza
+in the following folder:
+
+```
+_build/default/$path-to-directory-of-melange-emit-stanza/$target
```

Comment:
this is wrong, the path is:

```
_build/default/$path-to-directory-of-melange-emit-stanza/$path-to-target-inside-the-workspace/$target.js
```

Reply:
> this is wrong, the path is:
> `_build/default/$path-to-directory-of-melange-emit-stanza/$path-to-target-inside-the-workspace/$target.js`

I think that is not accurate either, as it's missing the `target` field from melange emit stanza.

It would be something like:

```
_build/default/$path-to-directory-of-melange-emit-stanza/$melange-emit-target-field-name/$path-to-module/$module.js
```

I revisited this part quite thoroughly, added a section "JavaScript artifacts layout", including a more complex example to illustrate how paths work and included your suggested generic "path to js":  https://github.com/melange-re/documentation-site/pull/25/commits/9e2c270da3c252425834b116a1069df7899f8b22